elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,245 @@
+that authentication has failed.
+
+If a user has previously been authenticated and their cookie has expired or the user is using the 
+admin console to connect to a different server it is possible the admin console may sent an invocation
+to the `/management` context without an appropriate cookie, in this case the admin console will need
+to be "notified to" and handle a redirect to the `/authentication` context.
+
+==== elytron-authentication
+
+A new subsystem `elytron-authentication` will be developed, the primary role of this subsystem
+will be to make a HTTP context available to be published.  This subsystem will also be responsible for
+the generation of the cryptographically signed JWT token and also provide a configured BEARER HTTP
+authentication mechanism which can validate the tokens produced by this server only.
+
+The generation and signing of the JWT tokens will largely be internal to the subsystem and will
+not be configurable.  The subsystem will manage a dynamically generated key pair for token signing
+and validation, this will live as long as the application server process and will not survive restarts.
+The key pair however will survive reloads as it is common when managing an application server to reload
+the configuration.
+
+Although this subsystem is only being used for the security of the management HTTP interface at this
+time later it will be expanded to be used for deployment security as well.
+
+The HTTP context configuration will cross reference a `security-domain` for the authentication 
+requirements.
+
+This subsystem will also support the definition of arbitrary attributes which can be requested from
+the user at the time of authentication, as an example this could be used to request a single use
+password in addition to the regular password.  A new evidence type will be defined within WildFly Elytron
+which will accept an arbitrary set of attributes during authentication.  The existing security realms
+within WildFly Elytron that handle username/password based authentication will be updated to support
+this new dynamic evidence type.
+
+==== core-management
+
+The number of attributes related to security on the management interfaces has grown with different
+permutations needed for different situations.  To reduce the complexity a new resource `policy=security`
+will be added to the `core-management` subsystem, this will allow references to the sasl or http 
+authentication factories as well as to ssl-contexts to be used.  This new policy resouce will also be
+able to reference the new context made available from the `elytron-authentication` subsystem.
+
+==== Management Interfaces
+
+On the management interfaces a new attribute `use-core-management-security-policy` will be added,
+this will be a boolean and when set to `true` the configuration defined on the `policy=security`
+resource will be used instead of the individual attributes on the management interface.
```

Comment:
This attribute design doesn't align well with how attributes work, i.e. x and y are alternatives and the kernel rejects setting both; it doesn't reject setting y if x is 'true'.

Reply:
We could avoid mutual exclusivity issues if we wanted by saying when set to true we use the items defined in policy=security unless also defined here so local references override central references.

Reply:
That can work if the details are consistent and will remain consistent.